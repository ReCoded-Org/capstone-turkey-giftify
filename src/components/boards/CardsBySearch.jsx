import React from "react";
import "./Board.css";
import { VscArrowRight } from "react-icons/vsc";
import { Container } from "react-bootstrap";
import { Link } from "react-router-dom";

function CardsBySearch({ allProducts, setItemsDetails, testSearchValue }) {
  return (
    <Container>
      <div className="row cards">
        {allProducts
          .filter((product) => {
            if (testSearchValue === "") {
              return null;
            } else if (
              product.productType
                .toLowerCase()
                .includes(testSearchValue.toLowerCase())
            ) {
              return product;
            } else if (
              product.productName
                .toLowerCase()
                .includes(testSearchValue.toLowerCase())
            ) {
              return product;
            } else if (
              product.ProductDescription.toLowerCase().includes(
                testSearchValue.toLowerCase()
              )
            ) {
              return product;
            } else if (
              product.itemCondition
                .toLowerCase()
                .includes(testSearchValue.toLowerCase())
            ) {
              return product;
            } else {
              return null;
            }
          })
          .map((filteredItem, index) => {
            return (
              <div key={index} className="oneCard col-3">
                <img
                  className="itemImage"
                  src={filteredItem.productImage}
                  alt={filteredItem.productName}
                />
                <p>
                  <b>{filteredItem.productName}</b>
                </p>
                <Link to="/productDetails">
                  <p
                    onClick={() => {
                      setItemsDetails(filteredItem);
                    }}
                  >
                    See Details <VscArrowRight />{" "}
                  </p>
                </Link>
              </div>
            );
          })}
      </div>
    </Container>
  );
}

export default CardsBySearch;
