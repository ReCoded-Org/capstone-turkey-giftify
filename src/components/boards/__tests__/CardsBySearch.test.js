import React from "react";
import CardsBySearch from "../CardsBySearch";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

const allProducts = [
  {
    productName: "bag",
    ProductDescription: "hand bag",
    itemCondition: "good",
    productType: "clothing",
    productImage: "image",
  },
  {
    productName: "book",
    ProductDescription: "turkish",
    itemCondition: "good",
    productType: "book",
    productImage: "image",
  },
];

const testSearchValue = "";

it("CardsBySearch rendered correctly", () => {
  const tree = renderer
    .create(
      <CardsBySearch
        allProducts={allProducts}
        testSearchValue={testSearchValue}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
