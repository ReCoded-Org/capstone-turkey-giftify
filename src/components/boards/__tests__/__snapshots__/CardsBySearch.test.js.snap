// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsBySearch rendered correctly 1`] = `
<div
  className="container"
>
  <div
    className="row cards"
  />
</div>
`;
